mm2src/crypto/src/key_derivation.rs

```diff
@@ -134,6 +134,11 @@ pub(crate) fn derive_keys_for_mnemonic(
 /// Splits a path into its components and derives a key for each component.
 fn derive_key_from_path(master_node: &[u8], path: &str) -> MmResult<[u8; 32], KeyDerivationError> {
     let mut current_key_material = master_node.to_vec();
+
+    if current_key_material.len() < 64 {
+        return MmError::err(KeyDerivationError::InvalidKeyLength);
+    }
+
```

Comment:
I think It's better if we do the check on master_node arg before assigning to current_key_material

Reply:
Done.